energy-integration/farm-boosted-yields/src/lib.rs

```diff
@@ -130,6 +187,45 @@ where
         ManagedVec::from_single_item(EsdtTokenPayment::new(reward_token_id, 0, total_rewards))
     }
 
+    // User rewards formula = user_base_rewards +
+    // min(base_const * user_base_rewards, boosted_rewards * (energy_const * user_energy / total_energy + farm_const * user_farm / total_farm) / (X+Y))
+    fn get_user_rewards_for_week(
+        &self,
+        module: &Self::WeeklyRewardsSplittingMod,
+        energy_amount: &BigUint<<Self::WeeklyRewardsSplittingMod as ContractBase>::Api>,
+        total_energy: &BigUint<<Self::WeeklyRewardsSplittingMod as ContractBase>::Api>,
+        total_rewards: &PaymentsVec<<Self::WeeklyRewardsSplittingMod as ContractBase>::Api>,
+    ) -> PaymentsVec<<Self::WeeklyRewardsSplittingMod as ContractBase>::Api> {
+        let mut user_rewards = ManagedVec::new();
+        let factors = module.boosted_yields_factors().get();
+        if energy_amount == &0
+            || energy_amount < &factors.min_energy_amount
+            || self.user_farm_amount < factors.min_farm_amount
+        {
+            return user_rewards;
+        }
+
+        // TODO
+        for weekly_reward in total_rewards {
+            // boosted_rewards * (energy_const * user_energy / total_energy + farm_const * user_farm / total_farm) / (X+Y)
+            let boosted_reward_amount = weekly_reward.amount
+                * ((&factors.user_rewards_energy_const * energy_amount / total_energy)
+                    + (&factors.user_rewards_farm_const * &self.user_farm_amount
+                        / &self.total_farm_supply))
+                / (&factors.user_rewards_energy_const + &factors.user_rewards_farm_const);
+
+            let normalized_user_base_rewards = BigUint::zero();
+            let user_base_rewards =
+                &factors.user_rewards_base_const * &normalized_user_base_rewards;
+            let user_reward = cmp::min(user_base_rewards, boosted_reward_amount);
+            if user_reward > 0 {
+                user_rewards.push(EsdtTokenPayment::new(weekly_reward.token_identifier, 0, user_reward));
```

Comment:
in the end these 2 payments will get burnt and simple lock contract will send the values to the user ?